### DISCUSSION TITLE
The WordPress Developer Survey

### DISCUSSION THREAD

**Discussion by jonathanbossenger** [WordPress/developer-blog-content#265] [Topic Ideas]
About a year ago I published a small survey of WordPress development tools on my personal blog: https://jonathanbossenger.com/2023/07/the-state-of-wordpress-developer-tools-survey-results/

I was thinking that this is something we could expand on, and use to survey all WordPress developers each year. This would help give us insights on various things like developer tooling, WordPress API experience, and knowledge of planned developer features.

This could be similar to the [Stack Overflow Developer Survey](https://survey.stackoverflow.co/2023/), but with a very specific focus on WordPress developers.

My suggestion is that we publish a post each year around June/July to survey all WordPress developers, and then report on the results around August/September of that year.

**Comment by crispopp** [comment]
About a year ago, I published a small survey on my personal blog about WordPress development tools: [The State of WordPress Developer Tools Survey Results]
I believe this is something we could expand and conduct annually to survey all WordPress developers. This would help us gain insights on various aspects such as developer tooling, WordPress API experience, and awareness of planned developer features.

Similar to the Stack Overflow Developer Survey, but with a specific focus on **[WordPress developers](https://caprahost.com/wordpress-hosting/)**, we could gather valuable data to understand trends, challenges, and preferences within the community.

I propose that we publish a post each year around June/July to conduct the survey, and then report on the results around August/September. This annual survey could become a go-to resource for anyone interested in the state of WordPress development.

**Comment by jonathanbossenger** [comment]
Next steps:

- [ ] What is the purpose of the survey? What is the survey trying to gather/learn/understand? What would a reader of the survey results gain from reading the results?
- [ ] What questions/topics would be included in the survey? What questions would not be included.

**Comment by jonathanbossenger** [comment]
## What is the purpose of this survey? What is the survey trying to gather/learn/understand?

The purpose of this survey is to understand the current state of the WordPress extender community (a.k.a. WordPress developers). 

These are the users of WordPress who build on top of WordPress every day, customizing WordPress core, plugins, and themes and building custom themes and plugins, either for their employers or for their customers. 

The survey would be modeled after the [Stack Overflow Developer Survey](https://survey.stackoverflow.co/2023/)

## What would a reader of the survey results gain from reading the results?

The WordPress project currently has three major platforms to speak directly to developers:
- The [developer resources site](https://developer.wordpress.org/)
- Any developer content on [Learn.WordPress.org](https://learn.wordpress.org/)
- The [WordPress developer blog ](https://developer.wordpress.org/news/)

Contributors to these platforms can use the results of this yearly survey to make better-informed decisions about new content that needs to be added (or perhaps updated) on these platforms to better serve the needs of the WordPress developer community.

Community members who read the results of this survey could use them to gauge where they fit into the extender community and learn about new and interesting tools, online learning resources, and developer communities to consider joining. It would also help promote WordPress developer-related content outside of the developer blog.

# What questions/topics would be included in the survey? What questions would not be included?

While not an extensive list, examples of topics could include things like:
- What development tools do you use daily?
- What industries do you work in?
- Are you a solo developer or working in a team? Freelance or employed?
- What projects do you work on/problems are you trying to solve with WordPress?
- To what developer communities do you belong?
- What developer conferences do you go to?
- Where do you go to learn about WordPress development?
- What types of things are you learning/do you need to learn about WordPress development?  

Additional questions about the specific developer platforms could include things like:
- What was your favorite/least favorite developer blog post of the past 12 months
- What lessons/courses would you like to see on Learn.WordPress.org
- What areas of the developer resources site need improvements?

**Comment by bph** [comment]
@jonathanbossenger Thank you so much for additional thoughts on the Developer survey. During last month's meeting, there was also a concern that a survey like this would need some cross-team collaboration and buy-in from project leadership as running of the survey would also require additional resources, for translation and analysis. 
I am also concerned about the insights we can glean from such a survey. "some things remain invisible in non-survey data: perceptions, knowledge and beliefs, attitudes, and reasoning." (Stefanie Stantcheva)

An example the question:  "What development tools do you use daily?" 
The outcome is a list of tools with some % - What would you learn from this you don't know already? 

"Are you a solo developer or working in a team? Freelance or employed?" 
The result would read like (with made up numbers)  15 % are freelancers, 25 % work in small team and 30 % work in larger teams and 25% at enterprise level? - Even if the numbers turn up to be slightly different, what would such a breakdown mean for your work and the content we would produce? 

The question: "What developer conferences do you go to?" seems a promising one. 
We would learn about conferences we might attend in addition to WordCamps to connect with other developers outside WordPress space. It might be a great to test how to best reach out to those developers...

**Comment by jonathanbossenger** [comment]
After various discussions, it looks like surveys are less and less beneficial, so I'm marking this as closed so we can consider other ways to source this information.
